File: AtCoder/cd_1183A.py
# CodeForce Problem 1183A

class Solution(object):
    def is_interesting(self,n:int)->bool:
        self.m = int()
        self.sum = 0
        while(n > 0):
            self.m = n%10
            self.sum = self.sum+self.m
            n//=10
        if(self.sum%4==0):
            return True
        return False


if __name__=='__main__':
    s = Solution()
    n = int(input())
    while(not s.is_interesting(n)):
        n+=1
    print(n)

File: AtCoder/cd_1183A.cxx
// CodeForce Problem 1183A

#include <bits/stdc++.h>
using namespace std;

class Solution{
	public:
	static bool is_interesting(int n){
		int m,sum = 0;
		while(n > 0){
			m = n%10;
			sum = sum+m;
			n=n/10;
		}
		return(sum%4==0 ? true:false);
	}
};

int main(){
	ios_base::sync_with_stdio(0);
	cin.tie(0); cout.tie(0);
	int n; cin >> n;
	while(!Solution::is_interesting(n)){
		n++;
	}
	cout << n;
	return 0;
}
